File: src/libqanystringview.cpp
#include <QAnyStringView>
#include <QByteArray>
#include <QChar>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qanystringview.h>
#include "libqanystringview.h"
#include "libqanystringview.hxx"

QAnyStringView* QAnyStringView_new(const libqt_string other) {
    QString other_QString = QString::fromUtf8(other.data, other.len);
    return new QAnyStringView(QAnyStringView(other_QString));
}

QAnyStringView* QAnyStringView_new2(libqt_string other) {
    QString other_QString = QString::fromUtf8(other.data, other.len);
    return new QAnyStringView(std::move(QAnyStringView(other_QString)));
}

QAnyStringView* QAnyStringView_new3() {
    return new QAnyStringView();
}

QAnyStringView* QAnyStringView_new4(const libqt_string str) {
    QByteArray str_QByteArray(str.data, str.len);
    return new QAnyStringView(str_QByteArray);
}

QAnyStringView* QAnyStringView_new5(const libqt_string str) {
    QString str_QString = QString::fromUtf8(str.data, str.len);
    return new QAnyStringView(str_QString);
}

QAnyStringView* QAnyStringView_new6(const libqt_string param1) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    return new QAnyStringView(QAnyStringView(param1_QString));
}

void QAnyStringView_CopyAssign(QAnyStringView* self, QAnyStringView* other) {
    *self = *other;
}

void QAnyStringView_MoveAssign(QAnyStringView* self, QAnyStringView* other) {
    *self = std::move(*other);
}

libqt_string QAnyStringView_Mid(const QAnyStringView* self, ptrdiff_t pos) {
    QAnyStringView _ret = self->mid((qsizetype)(pos));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Left(const QAnyStringView* self, ptrdiff_t n) {
    QAnyStringView _ret = self->left((qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Right(const QAnyStringView* self, ptrdiff_t n) {
    QAnyStringView _ret = self->right((qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Sliced(const QAnyStringView* self, ptrdiff_t pos) {
    QAnyStringView _ret = self->sliced((qsizetype)(pos));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Sliced2(const QAnyStringView* self, ptrdiff_t pos, ptrdiff_t n) {
    QAnyStringView _ret = self->sliced((qsizetype)(pos), (qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_First(const QAnyStringView* self, ptrdiff_t n) {
    QAnyStringView _ret = self->first((qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Last(const QAnyStringView* self, ptrdiff_t n) {
    QAnyStringView _ret = self->last((qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Chopped(const QAnyStringView* self, ptrdiff_t n) {
    QAnyStringView _ret = self->chopped((qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Slice(QAnyStringView* self, ptrdiff_t pos) {
    QAnyStringView _ret = self->slice((qsizetype)(pos));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string QAnyStringView_Slice2(QAnyStringView* self, ptrdiff_t pos, ptrdiff_t n) {
    QAnyStringView _ret = self->slice((qsizetype)(pos), (qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QAnyStringView_Truncate(QAnyStringView* self, ptrdiff_t n) {
    self->truncate((qsizetype)(n));
}

void QAnyStringView_Chop(QAnyStringView* self, ptrdiff_t n) {
    self->chop((qsizetype)(n));
}

libqt_string QAnyStringView_ToString(const QAnyStringView* self) {
    QString _ret = self->toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

ptrdiff_t QAnyStringView_Size(const QAnyStringView* self) {
    return static_cast<ptrdiff_t>(self->size());
}

const void* QAnyStringView_Data(const QAnyStringView* self) {
    return (const void*)self->data();
}

int QAnyStringView_Compare(libqt_string lhs, libqt_string rhs) {
    QString lhs_QString = QString::fromUtf8(lhs.data, lhs.len);
    QString rhs_QString = QString::fromUtf8(rhs.data, rhs.len);
    return QAnyStringView::compare(QAnyStringView(lhs_QString), QAnyStringView(rhs_QString));
}

bool QAnyStringView_Equal(libqt_string lhs, libqt_string rhs) {
    QString lhs_QString = QString::fromUtf8(lhs.data, lhs.len);
    QString rhs_QString = QString::fromUtf8(rhs.data, rhs.len);
    return QAnyStringView::equal(QAnyStringView(lhs_QString), QAnyStringView(rhs_QString));
}

QChar* QAnyStringView_Front(const QAnyStringView* self) {
    return new QChar(self->front());
}

QChar* QAnyStringView_Back(const QAnyStringView* self) {
    return new QChar(self->back());
}

bool QAnyStringView_Empty(const QAnyStringView* self) {
    return self->empty();
}

ptrdiff_t QAnyStringView_SizeBytes(const QAnyStringView* self) {
    return static_cast<ptrdiff_t>(self->size_bytes());
}

ptrdiff_t QAnyStringView_MaxSize(const QAnyStringView* self) {
    return static_cast<ptrdiff_t>(self->max_size());
}

bool QAnyStringView_IsNull(const QAnyStringView* self) {
    return self->isNull();
}

bool QAnyStringView_IsEmpty(const QAnyStringView* self) {
    return self->isEmpty();
}

ptrdiff_t QAnyStringView_Length(const QAnyStringView* self) {
    return static_cast<ptrdiff_t>(self->length());
}

libqt_string QAnyStringView_Mid2(const QAnyStringView* self, ptrdiff_t pos, ptrdiff_t n) {
    QAnyStringView _ret = self->mid((qsizetype)(pos), (qsizetype)(n));
    QString _qstr = _ret.toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _qstr.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc((_str.len + 1) * sizeof(char)));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

int QAnyStringView_Compare3(libqt_string lhs, libqt_string rhs, int cs) {
    QString lhs_QString = QString::fromUtf8(lhs.data, lhs.len);
    QString rhs_QString = QString::fromUtf8(rhs.data, rhs.len);
    return QAnyStringView::compare(QAnyStringView(lhs_QString), QAnyStringView(rhs_QString), static_cast<Qt::CaseSensitivity>(cs));
}

void QAnyStringView_Delete(QAnyStringView* self) {
    delete self;
}

File: include/libqanystringview.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qnamespace_enums = @import("libqnamespace.zig").enums;
const std = @import("std");

/// https://doc.qt.io/qt-6/qanystringview.html
pub const qanystringview = struct {
    /// New constructs a new QAnyStringView object.
    ///
    /// ``` other: []const u8 ```
    pub fn New(other: []const u8) QtC.QAnyStringView {
        const other_str = qtc.struct_libqt_string{
            .len = other.len,
            .data = other.ptr,
        };

        return qtc.QAnyStringView_new(other_str);
    }

    /// New2 constructs a new QAnyStringView object and invalidates the source QAnyStringView object.
    ///
    /// ``` other: []const u8 ```
    pub fn New2(other: []const u8) QtC.QAnyStringView {
        const other_str = qtc.struct_libqt_string{
            .len = other.len,
            .data = other.ptr,
        };

        return qtc.QAnyStringView_new2(other_str);
    }

    /// New3 constructs a new QAnyStringView object.
    ///
    ///
    pub fn New3() QtC.QAnyStringView {
        return qtc.QAnyStringView_new3();
    }

    /// New4 constructs a new QAnyStringView object.
    ///
    /// ``` str: []u8 ```
    pub fn New4(str: []u8) QtC.QAnyStringView {
        const str_str = qtc.struct_libqt_string{
            .len = str.len,
            .data = str.ptr,
        };

        return qtc.QAnyStringView_new4(str_str);
    }

    /// New5 constructs a new QAnyStringView object.
    ///
    /// ``` str: []const u8 ```
    pub fn New5(str: []const u8) QtC.QAnyStringView {
        const str_str = qtc.struct_libqt_string{
            .len = str.len,
            .data = str.ptr,
        };

        return qtc.QAnyStringView_new5(str_str);
    }

    /// New6 constructs a new QAnyStringView object.
    ///
    /// ``` param1: []const u8 ```
    pub fn New6(param1: []const u8) QtC.QAnyStringView {
        const param1_str = qtc.struct_libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };

        return qtc.QAnyStringView_new6(param1_str);
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: QtC.QAnyStringView, other: QtC.QAnyStringView ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QAnyStringView_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: QtC.QAnyStringView, other: QtC.QAnyStringView ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QAnyStringView_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#mid)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, allocator: std.mem.Allocator ```
    pub fn Mid(self: ?*anyopaque, pos: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Mid(@ptrCast(self), @intCast(pos));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Mid: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#left)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64, allocator: std.mem.Allocator ```
    pub fn Left(self: ?*anyopaque, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Left(@ptrCast(self), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Left: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#right)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64, allocator: std.mem.Allocator ```
    pub fn Right(self: ?*anyopaque, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Right(@ptrCast(self), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Right: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#sliced)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, allocator: std.mem.Allocator ```
    pub fn Sliced(self: ?*anyopaque, pos: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Sliced(@ptrCast(self), @intCast(pos));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Sliced: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#sliced)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, n: i64, allocator: std.mem.Allocator ```
    pub fn Sliced2(self: ?*anyopaque, pos: i64, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Sliced2(@ptrCast(self), @intCast(pos), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Sliced2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#first)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64, allocator: std.mem.Allocator ```
    pub fn First(self: ?*anyopaque, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_First(@ptrCast(self), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.First: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#last)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64, allocator: std.mem.Allocator ```
    pub fn Last(self: ?*anyopaque, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Last(@ptrCast(self), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Last: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#chopped)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64, allocator: std.mem.Allocator ```
    pub fn Chopped(self: ?*anyopaque, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Chopped(@ptrCast(self), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Chopped: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#slice)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, allocator: std.mem.Allocator ```
    pub fn Slice(self: ?*anyopaque, pos: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Slice(@ptrCast(self), @intCast(pos));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Slice: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#slice)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, n: i64, allocator: std.mem.Allocator ```
    pub fn Slice2(self: ?*anyopaque, pos: i64, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Slice2(@ptrCast(self), @intCast(pos), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Slice2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#truncate)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64 ```
    pub fn Truncate(self: ?*anyopaque, n: i64) void {
        qtc.QAnyStringView_Truncate(@ptrCast(self), @intCast(n));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#chop)
    ///
    /// ``` self: QtC.QAnyStringView, n: i64 ```
    pub fn Chop(self: ?*anyopaque, n: i64) void {
        qtc.QAnyStringView_Chop(@ptrCast(self), @intCast(n));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#toString)
    ///
    /// ``` self: QtC.QAnyStringView, allocator: std.mem.Allocator ```
    pub fn ToString(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_ToString(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.ToString: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#size)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Size(self: ?*anyopaque) i64 {
        return qtc.QAnyStringView_Size(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#data)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Data(self: ?*anyopaque) ?*const anyopaque {
        return qtc.QAnyStringView_Data(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#compare)
    ///
    /// ``` lhs: []const u8, rhs: []const u8 ```
    pub fn Compare(lhs: []const u8, rhs: []const u8) i32 {
        const lhs_str = qtc.struct_libqt_string{
            .len = lhs.len,
            .data = lhs.ptr,
        };
        const rhs_str = qtc.struct_libqt_string{
            .len = rhs.len,
            .data = rhs.ptr,
        };
        return qtc.QAnyStringView_Compare(lhs_str, rhs_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#equal)
    ///
    /// ``` lhs: []const u8, rhs: []const u8 ```
    pub fn Equal(lhs: []const u8, rhs: []const u8) bool {
        const lhs_str = qtc.struct_libqt_string{
            .len = lhs.len,
            .data = lhs.ptr,
        };
        const rhs_str = qtc.struct_libqt_string{
            .len = rhs.len,
            .data = rhs.ptr,
        };
        return qtc.QAnyStringView_Equal(lhs_str, rhs_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#front)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Front(self: ?*anyopaque) QtC.QChar {
        return qtc.QAnyStringView_Front(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#back)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Back(self: ?*anyopaque) QtC.QChar {
        return qtc.QAnyStringView_Back(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#empty)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Empty(self: ?*anyopaque) bool {
        return qtc.QAnyStringView_Empty(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#size_bytes)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn SizeBytes(self: ?*anyopaque) i64 {
        return qtc.QAnyStringView_SizeBytes(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#max_size)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn MaxSize(self: ?*anyopaque) i64 {
        return qtc.QAnyStringView_MaxSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#isNull)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn IsNull(self: ?*anyopaque) bool {
        return qtc.QAnyStringView_IsNull(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#isEmpty)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn IsEmpty(self: ?*anyopaque) bool {
        return qtc.QAnyStringView_IsEmpty(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#length)
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn Length(self: ?*anyopaque) i64 {
        return qtc.QAnyStringView_Length(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#mid)
    ///
    /// ``` self: QtC.QAnyStringView, pos: i64, n: i64, allocator: std.mem.Allocator ```
    pub fn Mid2(self: ?*anyopaque, pos: i64, n: i64, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAnyStringView_Mid2(@ptrCast(self), @intCast(pos), @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qanystringview.Mid2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#compare)
    ///
    /// ``` lhs: []const u8, rhs: []const u8, cs: qnamespace_enums.CaseSensitivity ```
    pub fn Compare3(lhs: []const u8, rhs: []const u8, cs: i64) i32 {
        const lhs_str = qtc.struct_libqt_string{
            .len = lhs.len,
            .data = lhs.ptr,
        };
        const rhs_str = qtc.struct_libqt_string{
            .len = rhs.len,
            .data = rhs.ptr,
        };
        return qtc.QAnyStringView_Compare3(lhs_str, rhs_str, @intCast(cs));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qanystringview.html#dtor.QAnyStringView)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QAnyStringView ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAnyStringView_Delete(@ptrCast(self));
    }
};
